pkg/types/spec/validations.go

```diff
@@ -1282,17 +1303,78 @@ func validateDockerImagePath(image string, providerType types.ProviderType, awsC
 				return nil
 			}
 
+			return err
+		}
+	} else if k8sClient != nil {
+		dockerAuthStr, err = getDockerAuthStrFromK8s(dockerClient, k8sClient)
+		if err != nil {
 			return err
 		}
 	}
 
-	if err := docker.CheckImageAccessible(dockerClient, image, dockerAuth); err != nil {
+	if err := docker.CheckImageAccessible(dockerClient, image, dockerAuthStr, providerType); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func getDockerAuthStrFromK8s(dockerClient *docker.Client, k8sClient *k8s.Client) (string, error) {
+	secretData, err := k8sClient.GetSecretData(_dockerPullSecretName)
+	if err != nil {
+		return "", err
+	}
+
+	// check if the user provided the registry auth secret
+	if secretData == nil {
+		return docker.NoAuth, nil
+	}
+
+	authData, ok := secretData[".dockerconfigjson"]
+	if !ok {
+		return "", ErrorUnexpectedDockerSecretData("should contain \".dockerconfigjson\" key", secretData)
+	}
+
+	var authSecret docker.K8sAuthSecret
+	err = json.Unmarshal(authData, &authSecret)
```

Comment:
Was the default json package used intentionally here as opposed to libjson?

Reply:
it was not intentional, thanks!